api/client/run.go

```diff
@@ -121,8 +121,8 @@ func (cli *DockerCli) initSpecialVolumes(config *container.Config, hostConfig *c
 	var (
 		initConfig     *container.Config
 		initHostConfig *container.HostConfig
-		execJobs       []string
-		execID         string
+		errCh          chan error
+		execCount      uint32
```

Comment:
WaitGroup has to wait for all ExecCmd goroutines to stop before checking results, while current implementation can check for results as it goes and returns failure as soon as any command fails.